packages/frontend/component/src/ui/masonry/masonry-item.tsx

```diff
@@ -0,0 +1,67 @@
+import clsx from 'clsx';
+import { useContext, useEffect, useMemo } from 'react';
+
+import { MasonryContext, type MasonryItem as MasonryItemType } from './context';
+import * as styles from './styles.css';
+export interface MasonryItemProps
+  extends Omit<React.HTMLAttributes<HTMLDivElement>, 'id' | 'height'>,
+    MasonryItemType {
+  locateMode?: 'transform' | 'leftTop' | 'transform3d';
+}
+
+export const MasonryItem = ({
+  id,
+  height,
+  className,
+  children,
+  locateMode = 'leftTop',
+  style: styleProp,
+  ...props
+}: MasonryItemProps) => {
+  const { addItem, removeItem, layoutMap, sleepMap } =
+    useContext(MasonryContext);
+
+  useEffect(() => {
+    addItem({ id, height });
+    return () => {
+      removeItem(id);
+    };
+  }, [addItem, height, id, removeItem]);
```

Comment:
Shall we put this into the top level context/config instead? It will cause quite some unnecessary updates in this child component.

Reply:
I found it hard to avoid rerendering with this api:

```tsx
<Masonry.Root>
  <Masonry.Item />
  <Masonry.Item />
  <Masonry.Item />
</Masonry.Root>
```

The MasonryItem will rerender when `sleepMap` recalculated, even if its state not changed. So I changed the api to:

```tsx
const items = useMemo(() => {
    return {
        id: '',
        height: 100,
        children: <div></div>
    }
}, [])
<Masonry items={[]} />
```

So we can pass `sleep` state as props